src/Symfony/Component/Lock/Serializer/LockKeyNormalizer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Serializer;
+
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+class LockKeyNormalizer implements NormalizerInterface, DenormalizerInterface
+{
+    public function getSupportedTypes(?string $format): array
+    {
+        return [Key::class => true];
+    }
+
+    /**
+     * @param Key $data
+     */
+    public function normalize(mixed $data, ?string $format = null, array $context = []): array
+    {
+        return (fn () => array_intersect_key(get_object_vars($this), array_flip($this->__sleep())))->bindTo($data, Key::class)();
+    }
+
+    public function supportsNormalization(mixed $data, ?string $format = null, array $context = []): bool
+    {
+        return $data instanceof Key;
+    }
+
+    public function denormalize(mixed $data, string $type, ?string $format = null, array $context = []): Key
+    {
+        $key = (new \ReflectionClass(Key::class))->newInstanceWithoutConstructor();
+        foreach ($key->__sleep() as $serializedField) {
+            (fn () => $this->$serializedField = $data[$serializedField])->bindTo($key, Key::class)();
```

Comment:
it should be more efficient to have one closure outside of the loop instead of many calls inside

Reply:
Thanks, done.

Reply:
Read your comment again after 4 months – the loop is now inside the closure :+1: :smile: